prediction_market_agent/agents/replicate_to_omen_agent/omen_replicate.py

```diff
@@ -132,17 +133,28 @@ def omen_replicate_from_tx(
         # Do as the last steps, because it calls OpenAI (costly & slow).
         if is_invalid(market.question):
             logger.info(
-                f"Skipping `{market.question}` because it seems to be an invalid question."
+                f"Skipping `{market.question}` was marked as invalid. Trying to rephrase and make it valid."
             )
-            continue
+            new_question = rephrase(market.question)
+            logger.info(f"Rephrased `{market.question}` to `{new_question}`.")
+
+            # Reinsert at the beginning of the list to reprocess with the new question
+            if is_invalid(new_question):
+                logger.info(
+                    f"Skipping `{new_question}` because it could not be rephrased into a valid question."
+                )
+                continue
+            else:
+                market.question = new_question
 
         if not is_predictable_binary(market.question):
             logger.info(
                 f"Skipping `{market.question}` because it seems to not be predictable."
             )
             continue
 
-        if market.description and not is_predictable_without_description(
+        # We don't check this for Polymarket because, if is_predictable is false, we rephrase the question.
```

Comment:
It seems to me form code above, that Manifold questions will get rephrased as well.

I think that will happen quite often. So maybe here something like `if was_rephrased and market.description and not ...`, instead of checking for manifold?

Reply:
Please check https://github.com/gnosis/prediction-market-agent/pull/912/commits/53a87eba5fbad79dafc31d66725373cdb551871c#diff-7706eb1de513aca7a2f2070c1c3dfa6488a7aaef052cff121b96aa7292da3047R180-R186

Now I'm skipping this for the case when it `was_rephrased`, because I run a check previously making sure it's predictable, so this check didn't seem necessary.